packages/slice-machine/components/ScreenshotChangesModal/index.tsx

```diff
@@ -0,0 +1,205 @@
+import Card from "@components/Card";
+import { AiOutlinePicture } from "react-icons/ai";
+import { RiErrorWarningLine } from "react-icons/ri";
+import SliceMachineModal from "@components/SliceMachineModal";
+
+import { useSelector } from "react-redux";
+import { isModalOpen } from "@src/modules/modal";
+import { SliceMachineStoreType } from "@src/redux/type";
+import { ModalKeysEnum } from "@src/modules/modal/types";
+
+import { Flex, Heading, Close, Box, Text } from "theme-ui";
+
+import { ComponentUI } from "@lib/models/common/ComponentUI";
+import { useState } from "react";
+
+const VariationIcon: React.FC<{ isValid?: boolean }> = ({ isValid }) => (
```

Comment:
put those in different files please, 1 component 1 file.
They can all be scoped in the ScreenshotChangesModal folder.

Reply:
No, "1 component 1 file" is not a rule you should comply with. See this tweet from Dan Abramov and many other articles: https://twitter.com/dan_abramov/status/1439304400464449536?s=20&t=7Xi2iln4sENglrouwI5RWA. VariationIcon is solely used in that component and is split for convenience.

Reply:
I use that rule for code readability.
Big files are hard to read, if this is used only here, then i'd create a new file in the same folder, it won't be used anywhere else but the code would be scoped and easier to read.

This could prevent big chunks of imports in one file and sometimes makes it easier when a fix is to make to find the right component to make the fix on.

Reply:
I don’t know what are your convention on SM but either you should make sure that either a component remain in the same file if used only specifically in one context or if it’s too big you can split but you need to make sure that it cannot be used elsewhere. By exporting it you take that risk.

Reply:
I definitely agree. VariationIcon is only used there and is mostly a way to split huge chunks of JSX